app/utility/base_planning_svc.py

```diff
@@ -78,10 +79,8 @@ async def remove_completed_links(operation, agent, links):
         :param agent:
         :return: updated list of links
         """
-        completed_links = [(l.command_hash if l.command_hash else l.command) for l in operation.chain
-                           if l.paw == agent.paw and (l.finish or l.can_ignore())]
-        return [l for l in links if l.ability.repeatable or
-                (l.command_hash if l.command_hash else l.command) not in completed_links]
+        completed_links = [lnk for lnk in operation.chain if lnk.paw == agent.paw and (lnk.finish or lnk.can_ignore())]
+        return [lnk for lnk in links if lnk.ability.repeatable or lnk not in completed_links]
```

Comment:
This change had to be made in combination with the override of Link equality for when a lateral movement Link has already been run. 

Without this change, when creating new lateral movement Link variants, every Link will be considered "not completed" because the replaced `origin_link_id` in the Link's `command` or Link's `command_hash` will be different for every Link created. This results in repeated lateral movements to the same host because the server does not believe it has run this lateral movement command yet. Therefore, the comparison of a Link's `command` or `command_hash` is not suitable to filter out completed links with `origin_link_id` in the commands.

This solution resolves this issue by taking advantage of overriding the [Link equality](https://github.com/mitre/caldera/pull/1961/files#diff-9ca0d40ddf617eaafbb77225dd5424ea5c732e93c60da859ae3b5c2bedec0a87R109), ensuring that if (1) the agent that the Link is being run on is the same, (2) the ability ID being run is the same, and (3) the used facts in the Link are the same, then the Links are equal, regardless of the Link's command or command hash.